crypto/bls/src/impls/blst.rs

```diff
@@ -157,10 +165,9 @@ impl TSignature<blst_core::PublicKey> for blst_core::Signature {
     }
 
     fn verify(&self, pubkey: &blst_core::PublicKey, msg: Hash256) -> bool {
-        if !self.subgroup_check() {
-            return false;
-        }
```

Comment:
I think with this subgroup check removed from `PublicKey` deserialisation we need to make sure we check this at the point new pubkeys are introduced into the system, namely in `process_deposit`:

https://github.com/sigp/lighthouse/blob/c6baa0eed131c5e8ecc5860778ffc7d4a4c18d2d/consensus/state_processing/src/per_block_processing.rs#L460-L464

Does that sounds right to you @kirk-baird ?

Reply:
The `PublicKey` subgroup check is being done at deserialisation time in `key_validate()`

https://github.com/sean-sn/lighthouse/blob/blst-subgroup-changes/crypto/bls/src/impls/blst.rs#L126-L135

So we don't need to do anything extra for `PublicKeys`.

The line you've highlighted above is the the subgroup check for the signature, which is still done it's just been moved :P the first variable of `self.verify()` is `true` and this will do the subgroup check in BLST.

https://github.com/sean-sn/lighthouse/blob/blst-subgroup-changes/crypto/bls/src/impls/blst.rs#L170

Reply:
Oh yes! Right you are, for some reason I thought I was looking at the pubkey